### PR TITLE
feat(RHICOMPL-1360): Add clowder config conditional for database config

### PR DESCRIPTION
**Pull request by Victoremepunto**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
+  else
+
+    db_service = ENV.fetch("DATABASE_SERVICE_NAME","").upcase.sub("-", "_")
+
+    username = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? "postgres" : ENV["POSTGRESQL_USER"]
+    password = ENV.key?("POSTGRESQL_ADMIN_PASSWORD") ? ENV["POSTGRESQL_ADMIN_PASSWORD"] : ENV["POSTGRESQL_PASSWORD"]
+    host = ENV.fetch("#{db_service}_SERVICE_HOST")
+    database = Rails.env.test? ? ENV['POSTGRESQL_TEST_DATABASE'] || ENV["POSTGRESQL_DATABASE"] : ENV["POSTGRESQL_DATABASE"]
+  end
+ %>
```
Comment (akofink) [inline config/database.yml:19]: This... shouldn't work? These are local variables defined within the if/else scope and should be not defined below when you're using them. You could keep the existing three lines at the top and just override the variables if clowder is enabled.

Comment (vkrizan) [reply config/database.yml:19]: These variables are all within the scope of the YAML/whole template. Ruby doesn't scope within if or begin blocks, as for example JS does with `let`.

Comment (akofink) [reply config/database.yml:19]: Hmm! All this time, I didn't realize that :laughing:

File: config/database.yml [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
   password: <%= password %>
-  host: <%= ENV["#{db_service}_SERVICE_HOST"] %>
-  database: <%= ENV["POSTGRESQL_DATABASE"] %>
+  host: <%= host %>
+  database: <%= database %>
 
 test:
   <<: *default
-  database: <%= ENV['POSTGRESQL_TEST_DATABASE'] || ENV['POSTGRESQL_DATABASE'] %>
+  database: <%= database %>
```
[the remark this answers is not in the document]
Comment (Victoremepunto) [reply config/database.yml:32]: I think you're right, thanks for spotting this :)

**Pull request merged**